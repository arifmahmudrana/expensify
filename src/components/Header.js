import React from 'react';
import {Link} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">  
    <div className="wrapper header__container">
      <Link className="header__title" to="/dashboard">
        <h1>ExpensiFy</h1>
      </Link>
      <button className="btn btn--link" onClick={startLogout}>Logout</button>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);