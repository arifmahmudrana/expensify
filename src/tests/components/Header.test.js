import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('header should render correctly', () => {
  const wrapper = shallow(<Header />);

  expect(wrapper).toMatchSnapshot();
});