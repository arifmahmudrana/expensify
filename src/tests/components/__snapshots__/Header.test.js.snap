// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render correctly 1`] = `
<header
  className="header"
>
  <div
    className="wrapper header__container"
  >
    <Link
      className="header__title"
      replace={false}
      to="/dashboard"
    >
      <h1>
        ExpensiFy
      </h1>
    </Link>
    <button
      className="btn btn--link"
      onClick={[MockFunction]}
    >
      Logout
    </button>
  </div>
</header>
`;
